File: articles/behavioral-economics-organizational-decision-making/fortran/organizational_decision_model.f90
program organizational_decision_model
  implicit none

  real :: expected_payoff, risk, sunk_cost, prestige_value
  real :: complexity, overconfidence, review_strength
  real :: short_term_pressure, long_horizon_value, long_horizon_weight
  real :: value, probability

  expected_payoff = 0.14
  risk = 0.22
  sunk_cost = 0.31
  prestige_value = 0.20
  complexity = 0.35
  overconfidence = 0.18
  short_term_pressure = 0.70
  review_strength = 0.85
  long_horizon_value = 0.26
  long_horizon_weight = 0.60

  value = expected_payoff &
        + prestige_value * short_term_pressure &
        - risk &
        - complexity &
        + 0.9 * sunk_cost &
        + 0.7 * overconfidence &
        - 0.8 * review_strength * sunk_cost &
        - 0.5 * review_strength * overconfidence &
        + long_horizon_weight * long_horizon_value

  probability = 1.0 / (1.0 + exp(-value))

  print *, "Synthetic approval probability:", probability
end program organizational_decision_model

File: articles/behavioral-economics-organizational-decision-making/go/organizational_decision_model.go
package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func approvalProbability(
	expectedPayoff float64,
	risk float64,
	sunkCost float64,
	prestigeValue float64,
	complexity float64,
	overconfidence float64,
	shortTermPressure float64,
	reviewStrength float64,
	longHorizonValue float64,
	longHorizonWeight float64,
) float64 {
	value := expectedPayoff +
		prestigeValue*shortTermPressure -
		risk -
		complexity +
		0.9*sunkCost +
		0.7*overconfidence -
		0.8*reviewStrength*sunkCost -
		0.5*reviewStrength*overconfidence +
		longHorizonWeight*longHorizonValue

	return logistic(value)
}

func main() {
	p := approvalProbability(0.14, 0.22, 0.31, 0.20, 0.35, 0.18, 0.70, 0.85, 0.26, 0.60)
	fmt.Printf("Synthetic approval probability: %.3f\n", p)
}
